app/views/clients/_client_list.haml

```diff
@@ -68,9 +70,19 @@
                 %nobr
                   = masked_ssn client.ssn
                   = client_data_quality client, :ssn
-              %th.text-center
+              %td.text-center
                 = checkmark client.veteran
-              %th.text-center
+              %td.text-center
                 = checkmark client.chronic_homeless
+              %td
+                - if client.project_client.data_source_id == non_hmis_source_id
+                  - if client.project_client.is_deidentified?
+                    = link_to edit_deidentified_client_path(client.project_client.id_in_data_source), class: ['btn', 'btn-sm', 'btn-secondary'] do
```

Comment:
The logic for where the link goes can be moved up into the `NonHmisClient` models in `#edit_path`.  Each should return arrays that would work with `polymorphic_path` so we can move the complexity out of the view.

Reply:
I was able to leverage existing polymorphism in the NonHmis controller to achieve the same effect.